src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -95,6 +95,7 @@ def _validate_whl_extension(ext_file):
 
 
 def _add_whl_ext(source, ext_sha256=None, pip_extra_index_urls=None, pip_proxy=None):  # pylint: disable=too-many-statements
+    from humanfriendly import AutomaticSpinner
```

Comment:
You should getting the get a hand hold of `cli_ctx` and use `cli_ctx.get_progress_controller`

Reply:
@yugangw-msft  Resolved the comment. Please take look. Thanks!